### PR TITLE
Feature/cice6update

### PR DESCRIPTION
**Pull request by jkuang-exploro**
Integrate CICE6 into feature/coupled-crow

### CODE REVIEW

**Comment by JessicaMeixner-NOAA** [comment]
@JianKuang-NOAA you mentioned in slack that just the mediator cold start was running.  Is everything running at this point or is this to just review the progress to now?

**Comment by jkuang-exploro** [comment]
@JessicaMeixner-NOAA I was about to send you an email explaining it. The medcold is completed and highres forecast is undergoing. I think it a good idea to start a DRAFT PR and start inviting people to try it out.

File: ush/parsing_namelists_CICE.sh [hunk 1]
```diff
@@ -19,12 +19,15 @@ cat > ice_in <<eof
   , use_restart_time = $USE_RESTART_TIME
   , restart_format = 'nc'
   , lcdf64         = .false.
-  , restart_dir    = './restart/'
+  , numin          = 21
```
Comment (JessicaMeixner-NOAA) [inline ush/parsing_namelists_CICE.sh]: Should we go ahead and update the format here be updated to match that of https://github.com/ufs-community/ufs-s2s-model/blob/s2s_prototype5.0/parm/ice_in_template ? so no more , at the beginning of every line?

File: modulefiles/module_base.orion [hunk 2]
```diff
@@ -50,7 +50,7 @@ module load prod_util/1.2.0
 ##
 module use /apps/contrib/NCEPLIBS/lib/modulefiles
 module load netcdfp/4.7.4.release
-module load esmflocal/8.1.0.21bs.release
+module load esmflocal/8.1.0.27bs.release
```
Comment (JessicaMeixner-NOAA) [inline modulefiles/module_base.orion:53]: can we also update this for module_base.hera?

Comment (jkuang-exploro) [reply modulefiles/module_base.orion:53]: Got it! I will.

File: ush/parsing_namelists_CICE.sh (hunk 1 above)
Comment (jkuang-exploro) [reply ush/parsing_namelists_CICE.sh]: This has been addressed

File: ush/parsing_namelists_CICE.sh [hunk 3]
```diff
@@ -4,371 +4,594 @@ CICE_namelists(){
...
-  , processor_shape   = 'slenderX2'
-  , distribution_type = 'cartesian'
-  , distribution_wght = 'latitude'
-  , ew_boundary_type  = 'cyclic'
-  , ns_boundary_type  = 'tripole'
-  , maskhalo_dyn      = .false.
-  , maskhalo_remap    = .false.
-  , maskhalo_bound    = .false.
+   grid_format  = 'nc'
+   grid_type    = 'displaced_pole'
```
Comment (DeniseWorthen) [inline ush/parsing_namelists_CICE.sh]: This needs to be set 'tripole'

File: ush/parsing_namelists_CICE.sh [hunk 4]
```diff
@@ -4,371 +4,594 @@ CICE_namelists(){
...
+   f_strcorx       = 'x' 
+   f_strcory       = 'x' 
+   f_strocnx       = 'mdhxx' 
+   f_strocny       = 'mdhxx' 
+   f_strintx       = 'x' 
+   f_strinty       = 'x'
+   f_taubx         = 'x'
+   f_tauby         = 'x'
+   f_strength      = 'x'
+   f_divu          = 'mdhxx'
```
Comment (DeniseWorthen) [inline ush/parsing_namelists_CICE.sh]: I see that the settings for f_divu, f_shear, f_dvidtt, f_dvidtd, f_daitt, f_daidtd have been copied from the previous settings. I really can't imagine these being useful in the current context and I would remove them (change 'mdhxx' to 'x'. They are no longer saved in ufs-s2s-model.

File: ush/parsing_namelists_CICE.sh [hunk 5]
```diff
@@ -4,371 +4,594 @@ CICE_namelists(){
...
+   f_PPnet        = 'x'
+   f_algalpeak    = 'x'
+   f_zbgc_frac    = 'x'
 /
 
 &icefields_drag_nml
-    f_drag         = 'mdhxx'
-  , f_Cdn_atm      = 'mdhxx'
-  , f_Cdn_ocn      = 'mdhxx'
+   f_drag         = 'mdhxx'
```
Comment (DeniseWorthen) [inline ush/parsing_namelists_CICE.sh]: These have also been removed (changed to 'x') in ufs-s2s (f_drag, f_Cdn_atm, f_Cdn_ocn).

File: ush/parsing_namelists_CICE.sh (hunk 3 above)
Comment (jkuang-exploro) [reply ush/parsing_namelists_CICE.sh]: OK. I changed it

File: ush/parsing_namelists_CICE.sh (hunk 4 above)
Comment (jkuang-exploro) [reply ush/parsing_namelists_CICE.sh]: Sorry I missed that.

File: ush/parsing_namelists_CICE.sh (hunk 5 above)
Comment (jkuang-exploro) [reply ush/parsing_namelists_CICE.sh]: OK. I changed it.

**Comment by jiandewang** [comment]
compiling fine on HERA, first try failed with "state UNAVAILABLE" in waveinit step, second try failed with error 
JWAVE_INIT: line 43: setpdy.sh: command not found in waveinit step

**Comment by jkuang-exploro** [comment]
@jiandewang this indicate a problem of prod_util. Please consult Kate and Hang Lei for what happened for prod_util installation on Hera

**Comment by jiandewang** [comment]
@JianKuang-NOAA : I contacted Hang Lei and checked the library on HERA, prod_util contains setpdy.sh, so the problem is not in prod_util module

**Comment by jkuang-exploro** [comment]
@jiandewang Did you make sure that prod_util is loaded during runtime? I am not able to access Hera to check directly.

**Comment by jiandewang** [comment]
@JianKuang-NOAA yes it is

**Comment by jkuang-exploro** [comment]
@jiandewang Could you please manually load prod_util in command window and call setpdy.sh?

**Comment by jkuang-exploro** [comment]
If it's loaded successfully, then it would show these:

*************************************************************
** FATAL ERROR: cycle variable not set or formatted incorrectly
** ABNORMAL EXIT at Fri Sep 11 15:33:14 CDT 2020 on Orion-login-1.HPC.MsState.Edu
*************************************************************


Currently Loaded Modules:
  1) rocoto/1.3.2   2) contrib/0.1   3) prod_util/1.2.0

**Comment by jkuang-exploro** [comment]
If it's not, then it will simply show this:
Orion-login-1[204] jkuang$ setpdy.sh
-bash: setpdy.sh: command not found

**Comment by jiandewang** [comment]
@JianKuang-NOAA this is what I can see on HERA:
/home/Jiande.Wang[102] which setpdy.sh
/scratch2/NCEPDEV/nwprod/NCEPLIBS/utils/prod_util.v1.1.0/ush/setpdy.sh

/home/Jiande.Wang[103] setpdy.sh

*************************************************************
** FATAL ERROR: cycle variable not set or formatted incorrectly
** ABNORMAL EXIT at Fri Sep 11 16:48:52 EDT 2020 on hfe09
*************************************************************

**Comment by jiandewang** [comment]
@JianKuang-NOAA I think we need a third person to take a try to rule out setting issue.

**Comment by jkuang-exploro** [comment]
@jiandewang I agree. If you could share with me the log as a file, I could probably take a deeper look.

**Comment by jiandewang** [comment]
@JianKuang-NOAA see orion: /work/noaa/marine/Jiande.Wang/For-others/For-Jian

**Comment by jkuang-exploro** [comment]
@jiandewang I see the problem here:

On hera, I've set the version of esmf to be: "esmflocal/8.1.0.27bs.release", same as Orion.

However, this version is unavailable on Hera. As a result, module load process failed and no module is loaded at all.

Please check the proper esmf version on Hera for this workflow and let me know what it is.

Or, you could modify this file: modulefiles/module_base.hera directly.

**Comment by jiandewang** [comment]
@JianKuang-NOAA I put HERA esmf info at /work/noaa/marine/Jiande.Wang/For-others/For-Jian,

**Comment by jkuang-exploro** [comment]
@jiandewang Since I didn't find 27bs version as same as Orion, I temporally put esmf/8.0.1.08bs_ParallelNetCDF in the module file. Please try it though it may or may not work. Meanwhile, I encourage you to find another guy to tweak the modulefile. It doesn't make much sense to let someone without hera access to do this job, to be frankly.

**Comment by jiandewang** [comment]
@JianKuang-NOAA for HERA what you need is
module use -a /scratch1/NCEPDEV/nems/emc.nemspara/soft/modulefiles
module load netcdf_parallel/4.7.4.release
module load esmf/8.1.0bs27
module load hdf5_parallel/1.10.6.release

now I switch to Orion, compiling fine, cold start failed, see error in log file at /work/noaa/marine/Jiande.Wang/CROW/wrk-129/COMROOT

**Comment by jkuang-exploro** [comment]
@jiandewang I fixed the bug in config.fcst which causes the crash of your test. Meanwhile I updated the module file for Hera again.

**Comment by jiandewang** [comment]
@JianKuang-NOAA for hera module you had a typo
it should be module use -a /scratch1/NCEPDEV/nems/emc.nemspara/soft/modulefiles
you missed "use" here

**Comment by jiandewang** [comment]
@JianKuang-NOAA on Orion it passed fcst and post steps but wave post ended with non-zero return code. I think the wave post job generated what the nc files we need, but there is some grib2 related thing in the script which caused the final non zero exit code. See log file at /work/noaa/marine/Jiande.Wang/CROW/wrk-129/COMROOT/test_3d/logs/2013040100/gfs.wavepost.log around line 4420

**Comment by JessicaMeixner-NOAA** [comment]
wave post is a known issue - I'm working on it.

**Comment by jkuang-exploro** [comment]
@jiandewang @JessicaMeixner-NOAA I fixed the typo that Jiande mentioned. I suggest merge this PR and solve the wavepost issue in another PR once the Hera test pass.

**Review by DeniseWorthen — APPROVED** [review]
This looks good for the cice6 changes.

**Review by JessicaMeixner-NOAA — APPROVED** [review]
based on that @jiandewang can run on orion (and maybe hera too?) and @DeniseWorthen gave the go ahead

**Comment by jiandewang** [comment]
running on HERA now, need half hour for final confirmation

**Review by jiandewang — APPROVED** [review]

**Comment by jiandewang** [comment]
@JianKuang-NOAA remaining issue: (1) rocotoview does not work, (2) need to reduce post job #, will be hard to manage when run 35 days run

**Pull request merged**